src/components/scenes/PinLoginScene.tsx

```diff
@@ -87,20 +90,34 @@ export function PinLoginScene(props: Props) {
 
   // Runs once at start:
   React.useEffect(() => {
-    if (username && touch !== 'FaceID') {
-      dispatch(loginWithTouch(username)).catch(showError)
+    if (userInfo != null && touch !== 'FaceID') {
```

Comment:
Unless we need an undef userInfo for something else, I think it would be more
clear if we don't return an undefined userInfo in acb0ffb, and instead
explicitly return a userInfo with
{
  username: undefined,
  pinLoginEnabled: true,
  touchLoginEnabled: false
}